### PR TITLE
QOS fieldvalue reference ABNF format to string changes

### PR DESCRIPTION
**Pull request by AshokDaparthi**
Depends on https://github.com/Azure/sonic-utilities/pull/1626
Depends on https://github.com/Azure/sonic-buildimage/pull/7752

Qos tables in config db and app db used ABNF  format i.e "[TABLE_NAME|name] to refer fieldvalue other qos tables. 

Example: 
Config DB:
"Ethernet92|3": {
       "scheduler": "[SCHEDULER|scheduler.1]",
        "wred_profile": "[WRED_PROFILE|AZURE_LOSSLESS]"
  },
"Ethernet0|0": {
       "profile": "[BUFFER_PROFILE|ingress_lossy_profile]"
 },
  "Ethernet0": {
            "dscp_to_tc_map": "[DSCP_TO_TC_MAP|AZURE]",
            "pfc_enable": "3,4",
            "pfc_to_queue_map": "[MAP_PFC_PRIORITY_TO_QUEUE|AZURE]",
            "tc_to_pg_map": "[TC_TO_PRIORITY_GROUP_MAP|AZURE]",
            "tc_to_queue_map": "[TC_TO_QUEUE_MAP|AZURE]"
 },

AppDB: 
"BUFFER_QUEUE_TABLE:Ethernet88:3-4": {
      "profile": "[BUFFER_PROFILE_TABLE:egress_lossless_profile]"
  },

This format is not consistent with other DB schema followed in sonic. 
And also this reference in DB is not required, This is taken care by YANG "leafref". 

Removed this format and made it as string.
Example: 
 "Ethernet92|3": {
       "scheduler": "scheduler.1",
        "wred_profile": "AZURE_LOSSLESS"
  },

Dependent pul; requests: 
https://github.com/Azure/sonic-buildimage/pull/7752  - To modify platfrom files 
https://github.com/Azure/sonic-buildimage/pull/7281 - Yang model 
https://github.com/Azure/sonic-utilities/pull/1626   - DB migration

### CODE REVIEW

**Review by stephenxs — CHANGES_REQUESTED** [review]
Hi @AshokDaparthi 
I haven't got time to go over the entire PR. Some comments for now:
- Need to update the vs test cases as well, like [test_buffer_dynamic.py](https://github.com/Azure/sonic-swss/blob/master/tests/test_buffer_dynamic.py), [test_speed.py](https://github.com/Azure/sonic-swss/blob/master/tests/test_speed.py#L73), and [test_qos_map.py](https://github.com/Azure/sonic-swss/blob/master/tests/test_qos_map.py#L66)
I believe the current vs test is failing without these changes.
- Need to update the [db schema](https://github.com/Azure/sonic-swss/blob/master/doc/swss-schema.md) description.
- Can you fill the fields in the PR description?
 By doing so reviewers can have a better understanding of this PR.

Thanks.

**Comment by AshokDaparthi** [comment]
@stephenxs - Thanks  for information, i will dig to test cases. i compiled only for sonic-broadcom.bin , i do not see any failure at build time.

**Comment by stephenxs** [comment]
> @stephenxs - Thanks for information, i will dig to test cases. i compiled only for sonic-broadcom.bin , i do not see any failure at build time.

Yes. The vs test won't run when the sonic image (like sonic-broadcom.bin) is being built. It runs as part of the checks of the PR (`Azure.sonic-swss (Test vstest)`). To set up a local environment for vs test, you can refer [this guide](https://github.com/Azure/sonic-swss/blob/master/tests/README.md)

File: orchagent/orch.cpp [hunk 1]
```diff
@@ -310,66 +310,50 @@ bool Orch::bake()
...
+
+    if (ref_in.size() == 0)
     {
-        // value set by user is "[]"
+        // value set by user is ""
         // Deem it as a valid format
         // clear both type_name and object_name
         // as an indication to the caller that
         // such a case has been encountered
-        type_name.clear();
```
Comment (stephenxs) [inline orchagent/orch.cpp:335]: Can an empty string be passed by redis-db? If no and this is a valid flow, we need to find another way to indicate clearing reference. Probably we can use "NULL".

File: orchagent/orch.cpp [hunk 2]
```diff
@@ -301,7 +301,7 @@ bool Orch::bake()
 }
 
 /*
-- Validates reference has proper format which is [table_name:object_name]
+- Validates reference has proper format which is not ABNF [table_name:object_name]
```
Comment (stephenxs) [inline orchagent/orch.cpp:304]: Emphasizing `not ABNF` looks strange. I think `Validates reference has proper format which is object_name` should be good.

File: orchagent/orch.cpp [hunk 3]
```diff
@@ -310,66 +310,50 @@ bool Orch::bake()
...
-        type_name.clear();
         object_name.clear();
         return true;
     }
-    string ref_content = ref_in.substr(1, ref_in.size() - 2);
-    vector<string> tokens;
-    tokens = tokenize(ref_content, delimiter);
-    if (tokens.size() != 2)
+
+    if ((ref_in[0] == ref_start) || (ref_in[ref_in.size()-1] == ref_end))
```
Comment (stephenxs) [inline orchagent/orch.cpp:330]: Not sure whether we need to check whether it's surrounded by `[]` here.

File: cfgmgr/buffermgr.cpp [hunk 4]
```diff
@@ -284,14 +277,6 @@ void BufferMgr::doBufferTableTask(Consumer &consumer, ProducerStateTable &applTa
 
             for (auto i : kfvFieldsValues(t))
             {
-                SWSS_LOG_INFO("Inserting field %s value %s", fvField(i).c_str(), fvValue(i).c_str());
```
[the remark this answers is not in the document]
Comment (AshokDaparthi) [reply cfgmgr/buffermgr.cpp:287]: transformSeperator  is used to convert from config_db to app_db

File: cfgmgr/buffermgrdyn.cpp [hunk 5]
```diff
@@ -1951,8 +1939,7 @@ task_process_status BufferMgrDynamic::handleOneBufferPgEntry(const string &key,
             {
                 // Headroom override
                 pureDynamic = false;
-                transformReference(value);
```
[the remark this answers is not in the document]
Comment (AshokDaparthi) [reply cfgmgr/buffermgrdyn.cpp:1954]: Done

File: orchagent/orch.cpp (hunk 2 above)
Comment (AshokDaparthi) [reply orchagent/orch.cpp]: Done

**Review by stephenxs — APPROVED** [review]

**Review by stephenxs — APPROVED** [review]

File: tests/test_buffer_dynamic.py [hunk 6]
```diff
@@ -392,13 +392,13 @@ def test_mtuUpdate(self, dvs, testlog):
         self.app_db.wait_for_entry("BUFFER_PG_TABLE", "Ethernet0:3-4")
         self.app_db.wait_for_entry("BUFFER_PROFILE_TABLE", expectedProfileMtu)
         self.check_new_profile_in_asic_db(dvs, expectedProfileMtu)
-        self.app_db.wait_for_field_match("BUFFER_PG_TABLE", "Ethernet0:3-4", {"profile": "[BUFFER_PROFILE_TABLE:{}]".format(expectedProfileMtu)})
+        self.app_db.wait_for_field_match("BUFFER_PG_TABLE", "Ethernet0:3-4", {"profile": "{}".format(expectedProfileMtu)})
```
Comment (neethajohn) [inline tests/test_buffer_dynamic.py:395]: {"profile": expectedProfileMtu} should do

File: tests/test_buffer_dynamic.py [hunk 7]
```diff
@@ -392,13 +392,13 @@ def test_mtuUpdate(self, dvs, testlog):
...
 
         dvs.runcmd("config interface mtu Ethernet0 {}".format(default_mtu))
 
         self.app_db.wait_for_deleted_entry("BUFFER_PROFILE_TABLE", expectedProfileMtu)
         self.app_db.wait_for_entry("BUFFER_PROFILE_TABLE", expectedProfileNormal)
-        self.app_db.wait_for_field_match("BUFFER_PG_TABLE", "Ethernet0:3-4", {"profile": "[BUFFER_PROFILE_TABLE:{}]".format(expectedProfileNormal)})
+        self.app_db.wait_for_field_match("BUFFER_PG_TABLE", "Ethernet0:3-4", {"profile": "{}".format(expectedProfileNormal)})
```
Comment (neethajohn) [inline tests/test_buffer_dynamic.py:401]: {"profile": expectedProfileNormal} should do

File: tests/test_speed.py [hunk 8]
```diff
@@ -72,7 +72,7 @@ def test_SpeedAndBufferSet(self, dvs, testlog):
                 expected_pg_table = "Ethernet{}|3-4".format(i * 4)
                 assert expected_pg_table in pg_tables
 
-                expected_fields = {"profile": "[BUFFER_PROFILE|{}]".format(expected_new_profile_name)}
+                expected_fields = {"profile": "{}".format(expected_new_profile_name)}
```
Comment (neethajohn) [inline tests/test_speed.py:75]: {"profile": expected_new_profile_name} should do

**Review by neethajohn — COMMENTED** [review]
tests/test_buffer_traditional.py also requires changes

**Review by neethajohn — APPROVED** [review]

**Comment by zhangyanzhao** [comment]
@AshokDaparthi would you please check the build failure? Thanks.

**Comment by AshokDaparthi** [comment]
@stephenxs, @neethajohn - Please relook, i did modification to pass swss test cases until sonic-buildimage check in happens. Plan is to push ASAP and update sub modules.

**Comment by AshokDaparthi** [comment]
@neethajohn @stephenxs - Could you please help to review and merge, so that i can push other PR's

**Review by neethajohn — APPROVED** [review]

**Comment by prsunny** [comment]
@stephenxs , could you please sign-off?

**Review by stephenxs — APPROVED** [review]
LGTM.

**Comment by stephenxs** [comment]
> @stephenxs , could you please sign-off?

@prsunny approved.

**Comment by stephenxs** [comment]
> @stephenxs, @neethajohn - Please relook, i did modification to pass swss test cases until sonic-buildimage check in happens. Plan is to push ASAP and update sub modules.

Hi @AshokDaparthi 
Now the PR Azure/sonic-buildimage#7752 has been merged. Can you open a PR to remove the code added for passing the CI?
Thanks

**Comment by AshokDaparthi** [comment]
@stephenxs - Yes, i can do.  I am waiting for build to be ready with today sonic-buildimage changes. i did not see this yet in artifacts. only docker-sonic-vs.gz comes out of today image only checks will pass. So i will rise it tomorrow.

**Pull request merged**